spec/controllers/frontend_log_controller_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+describe FrontendLogController do
+  describe '#create' do
+    subject(:action) { post :create, params: params }
+
+    let(:user) { create(:user, :with_phone, with: { phone: '+1 (202) 555-1212' }) }
+    let(:params) { { event: 'custom event', payload: { message: 'To be logged...' } } }
+    let(:json) { JSON.parse(response.body, symbolize_names: true) }
+
+    context 'user is signed in' do
+
+      before do
+        sign_in user
+      end
+
+      it 'succeeds' do
+        action
+
+        expect(response).to have_http_status(:ok)
+        expect(json[:success]).to eq(true)
```

Comment:
can we add a `stub_analytics` and then some sort of `expect(@analytics).to have_received(:....).with(....)`

Reply:
yes! thanks!